src/huggingface_hub/hf_api.py

```diff
@@ -1724,17 +1735,78 @@ def move_repo(
             " completed."
         )
 
+    def create_commit(
+        self,
+        *,
+        repo_id: str,
+        operations: Iterable[CommitOperation],
+        commit_summary: str,
+        commit_description: Optional[str] = None,
+        token: Optional[str] = None,
+        repo_type: Optional[str] = None,
+        revision: Optional[str] = None,
+        create_pr: Optional[bool] = None,
+    ) -> Optional[str]:
```

Comment:
like it!

Reply:
looks nice! cc @gante who was looking for this kind of functionality too

Reply:
> I'm also not sure if create_pr should be a part of this method. Creating a PR seems to be more of its own operation to me

Good point, although passing a `create_pr` argument is close to the flow on the website (see the screenshot below)

![image](https://user-images.githubusercontent.com/33657802/171584614-dc02684c-2863-4c36-9266-4143da9854ac.png)

On the other hand, we could have the following API (which is closer to the "advanced" PR workflow)

```python
from huggingface_hub import create_pull_request, create_commit

revision = create_pull_request(...)
create_commit(..., revision=revision)
```

Reply:
Trying the website's flow, I find it quite confusing, since I expect to follow the same PR/MR workflow as GH or GitLab, but we seem to be forcing people on a branch name that we choose, and they then have to fetch that locally and then apply their changes.

The usual flow is that users create their branch locally, push, and then create their PR, and I'm not sure if we want to make people get used to a very different flow than what they already know.

Regardless, for users who are happy with their python code or terminal, it would make more sense to give them an interface which is more familiar to what they do in other projects/platforms.

Reply:
>  since I expect to follow the same PR/MR workflow as GH or GitLab

>  I'm not sure if we want to make people get used to a very different flow than what they already know.

> it would make more sense to give them an interface which is more familiar to what they do in other projects/platforms.

The reason functions such as `upload_file` exit in the hub library is because data scientists may not be familiar with the git ecosystem.

HF targets data scientists, whereas Github/Gitlab target software engineers.

We made the choice to be separate from Github/Gitlab (no "fork" button) because the use case is different. We just want the hub library to be easy for data scientists, with the least friction. It should make all the git commands (creating a PR, adding new stuff to a PR, ...) as simple as possible.

Reply:
Using git workflow has become quite normal even among data scientists in the industry. It might not be the case in academia, but in the industry that's how people work since they work with other dev teams which use the same workflow. I don't mind how it's implemented in the UI (although I don't like it), but in the library it should be something that more advanced users don't find unintuitive and confusing.

Reply:
We could maybe define a `create_pull_request` method that would be calling exactly `create_commit` with the `create_pr=True` param, what do you think?

Reply:
What's the input to `create_pull_request`? A target (and maybe source) branch name? Can we enable this workflow:

- `git checkout -b feature_branch`
- `create_commit`
- `create_pr(target='main')`

Reply:
no `create_pull_request` would be the exact same params as `create_commit` ie. a commit message and a changeset. 

No, we do not want to support PRs from/to arbitrary branches. It's a design decision, we can reassess in a year but I think it'll be a good design decision

Reply:
why?